check/requiretls/requiretls.go

```diff
@@ -0,0 +1,26 @@
+package requiretls
+
+import (
+	"github.com/foxcpp/maddy/check"
+	"github.com/foxcpp/maddy/exterrors"
+	"github.com/foxcpp/maddy/module"
+)
+
+func requireTLS(ctx check.StatelessCheckContext) module.CheckResult {
+	if ctx.MsgMeta.SrcTLSState.HandshakeComplete {
+		return module.CheckResult{}
+	}
+
+	return module.CheckResult{
+		Reason: &exterrors.SMTPError{
+			Code:         550,
+			EnhancedCode: exterrors.EnhancedCode{5, 7, 1},
+			Message:      "TLS conversation required",
+			CheckName:    "require_tls",
+		},
+	}
+}
+
+func init() {
+	check.RegisterStatelessCheck("require_tls", check.FailAction{Reject: true}, requireTLS, nil, nil, nil)
```

Comment:
I have a doubt about making this a per-connection check.

Say a connection is made, the client tries to send a message, and it fails because `require_tls` is enabled. Can the client then run `STARTTLS`? Will the per-connection check be re-run at that point, or will the original failure continue to apply? Should trhis, instead, be a per-message check, even though it's only checking connection params?

Reply:
STARTTLS resets the go-smtp session and the check will rerun.

Reply:
Though, CheckConnection is more of a logical rather than a technical separation. It still runs on the each message received over the same connections (though this might change in the future as an optimization).